tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,769 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/tensorflow/tensorflow/tensorflow/contrib/go/proto"
+)
+
+import "C"
+
+const (
+	cAckByte  = 6
+	cBellByte = 7
+	cDc1      = 17
+
+	cBytesComplex64 = 8
+	cBytesFloat32   = 4
+	cBytesFloat64   = 8
+	cBytesInt16     = 2
+	cBytesInt32     = 4
+	cBytesInt64     = 8
+	cBytesUint16    = 2
+)
+
+// A DataType represents the type of the data contained in a Tensor
+type DataType pb.DataType
+
+// A TensorInterface is the interface implemented by Tensors.
+type TensorInterface interface {
```

Comment:
In Go, interfaces are not prefixed by Interface.

If TensorInterface is representing a tensor, it should maybe be called just Tensor.